CMakeLists.txt

```diff
@@ -607,6 +607,16 @@ endif()
 
 if(IREE_BUILD_TESTS)
   enable_testing(iree)
+  set(IREE_REQUIRES_TMPDIR_TAG "requires-tmpdir")
+  set(IREE_TEST_TMPDIR "${IREE_BINARY_DIR}/test_tmpdir")
+  set(IREE_TEST_FAKE_TMPDIR "${IREE_BINARY_DIR}/not_a_test_tmpdir_you_should_not_write_here")
```

Comment:
This fake temporary directory isn't needed if we just give every test a directory. The idea was that if we want to only create directories for those tests that say they need it, tests that haven't said they need a tmp dir but try to use one should get clear errors. Creating a file and telling them it is a directory was the best way I found to do that.

Reply:
I like the simplicity of always creating a directory, as long as the overheads are low enough.

Reply:
Yeah, me too. It also matches the Bazel behavior. I just wanted to make sure it wasn't going to lead to too much overhead